tests/integration/connector.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+
+import mysql.connector
+import psycopg2
+from pymongo import MongoClient
+
+
+class MysqlConnector:
+    """Context manager for mysql connector."""
+
+    def __init__(self, config: dict):
+        """Initialize the context manager.
+
+        Args:
+            config: Configuration dict for the mysql connector, like:
+                config = {
+                    "user": user,
+                    "password": remote_data["password"],
+                    "host": host,
+                    "database": database,
+                    "raise_on_warnings": False,
+                }
+        """
+        self.config = config
+
+    def __enter__(self):
+        """Create the connection and return a cursor."""
+        self.connection = mysql.connector.connect(**self.config)
+        self.cursor = self.connection.cursor()
+        return self.cursor
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        """Handle transaction and connection close."""
+        self.connection.commit()
+        self.cursor.close()
+        self.connection.close()
+
+
+class PostgreSQLConnector:
+    """Context manager for PostrgreSQL connector."""
+
+    def __init__(self, config: str):
+        """Initialize the context manager.
+
+        Args:
+            config: Connection string for the postgresql connector
+        """
+        self.config = config
+
+    def __enter__(self):
+        """Create the connection and return a cursor."""
+        self.connection = psycopg2.connect(self.config)
+        self.connection.autocommit = True
```

Comment:
As said, I don't think that this context manager is really needed, provided psycopg2 already provides that functionality 

But if we want to keep it, I believe we should take the autocommit away. I thought we were using ContextManager for having unit of work, but this would commit the transaction straight away